stoken/main.py

```diff
@@ -48,8 +49,15 @@ def main(mode: str, encoding: str, variable_prefix: str, variable_suffix: str, d
         except Exception as e:
             quit_with_info(f"{Fore.RED}{e}\n\n"
                            f"Fail to read configuration file. Please refer to {README_URL} and modify the `{YAML_PATH}`.{colorama.Style.RESET_ALL}")
-
+    git_dev = BASE_DIR.resolve()
+    while not list(git_dev.glob("*.git")):
+        git_dev = git_dev.parent
+    print(git_dev)
+    repo = git.repo.base.Repo(git_dev)
```

Comment:
需要考虑`.git`不存在的情况 (git.repo.base.Repo实例化报错的情况)

```suggestion
    try:
        repo = git.repo.base.Repo(git_dev)
    except:
        repo = None
```